src/base/utils/fs.h

```diff
@@ -67,6 +67,40 @@ namespace Utils
         /* End of Qt4 code */
 
         QString cacheLocation();
+
+        /// Utility class to defer file deletion
+        class FileProcessingGuard
+        {
+        public:
+            FileProcessingGuard(const QString &path = QString());
+            ~FileProcessingGuard();
+
+            /// Cancels deferred file deletion
+            void releaseFile();
+
+        private:
+            QString m_path;
+        };
+
+        /// reads settings for .torrent files from preferences
+        /// and sets up file guard accordantly
+        class TorrentFileProcessignGuard : private FileProcessingGuard
```

Comment:
Then remove `AUTO_DELETE_` prefix from const names since it will become unneeded.
Also fix SETTINGS_KEY from `Downloads/` to `TorrentFileGuard/` since we use settings key names based on component names.